sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/Configs.java

```diff
@@ -360,6 +364,12 @@ public int getGlobalEndpointManagerMaxInitializationTimeInSeconds() {
         return getJVMConfigAsInt(GLOBAL_ENDPOINT_MANAGER_INITIALIZATION_TIME_IN_SECONDS, DEFAULT_GLOBAL_ENDPOINT_MANAGER_INITIALIZATION_TIME_IN_SECONDS);
     }
 
+    // Temporary. Thinclient endpoint discovery to be done through GetDatabaseAccount API
+    public URI getThinclientEndpoint() {
+        String uriString = System.getProperty("COSMOS.THINCLIENT_ENDPOINT");
```

Comment:
Please allow this to be configured via system property and environment variable - there are several other properties with this pattern recently added

Reply:
Same comment as above -

I thought we had previously discussed using one of the http2 flags as a switch for enabling thin client - I am ok to make this change but just want to get consensus on this taking this approach